app/uk/gov/hmrc/tariffclassificationfrontend/views/partials/sample/sample_status_events_table.scala.html

```diff
@@ -0,0 +1,58 @@
+@*
+ * Copyright 2019 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *@
+
+@import uk.gov.hmrc.tariffclassificationfrontend.config.AppConfig
+@import uk.gov.hmrc.tariffclassificationfrontend.models.request.AuthenticatedRequest
+@import uk.gov.hmrc.tariffclassificationfrontend.models._
+@import uk.gov.hmrc.tariffclassificationfrontend.utils.Dates
+@import uk.gov.hmrc.tariffclassificationfrontend.views.html.partials.{assignee, progressive_disclosure}
+@import uk.gov.hmrc.tariffclassificationfrontend.models.SampleStatusChange
+@(c: Case, events: Paged[Event])(implicit request: AuthenticatedRequest[_], messages: Messages, appConfig: AppConfig)
+
+<h3 class="heading-medium" id="sample-status-events-heading">Sample activity</h3>
+
+<table id="activity-events">
+    <thead>
+        <tr>
+            <th scope="col">Name</th>
+            <th scope="col">Status</th>
+            <th scope="col">Date</th>
+
+        </tr>
+    </thead>
+    <tbody>
+    @for((event: Event, index: Int) <- events.results.zipWithIndex) {
+        <tr id="sample-status-events-row-@{index}">
+            <td id="sample-status-events-row-@{index}-operator" class="sample-status-cell">@event.operator.name.getOrElse("Unknown")</td>
+
+            <td id="sample-status-events-row-@{index}-content" class="sample-status-cell-desc">
+                @{
+                    val body = event.details.asInstanceOf[SampleStatusChange]
```

Comment:
Not entirely sure if we should do it in this way.  The assumption that this events always will be a **SampleStatusChange** is a bit dangerous. Could we protected it with an **if**  ? maybe if you get a different one you would like to error or just pass

Reply:
I'll add a safety check around this, in same style as the table on activity view. This will throw a specific exception for unhandled types.